lib/moon/packages/data_bags/data_matrix.rb

```diff
@@ -152,7 +142,7 @@ def [](x, y, z)
       y = y.to_i
       z = z.to_i
       return @default unless contains?(x, y, z)
-      @data[calc_index(x, y, z)]
+      @data[calc_index(x, y, z)] || @default
```

Comment:
Not completely, the `@data` isn't fully initialized in order to have faster creations.

So the `@data` may just be an empty Array instead of a default filled Array.